src/modules/solution-explorer/solution-explorer-solution/solution-explorer-solution.ts

```diff
@@ -250,6 +253,48 @@ export class SolutionExplorerSolution {
     this._renameDiagramInput = input;
   }
 
+  public activateContextMenu(event: MouseEvent, diagram: IDiagram): void {
+    this._diagramInContextMenu = diagram;
+
+    const contextMenuOffset: number = 80;
+    this.diagramContextMenu.style.top = `${event.y - contextMenuOffset}px`;
+    this.diagramContextMenu.style.left = `${event.x}px`;
+    this.showContextMenu = true;
+
+    const documentEventListener: EventListenerOrEventListenerObject = (): void => {
+      this.showContextMenu = false;
+      this._diagramInContextMenu = undefined;
+
+      document.removeEventListener('click', documentEventListener);
+    };
+
+    document.addEventListener('click', documentEventListener);
+  }
+
+  public async copyDiagram(): Promise<void> {
+    if (!this._diagramInContextMenu) { return; }
+
+    let newNameFound: boolean = false;
+    let newName: string;
+    let diagramNumber: number = 1;
+
+    while (newNameFound === false) {
+      newName = `${this._diagramInContextMenu.name} (${diagramNumber})`;
+
+      newNameFound = this.openedDiagrams.every((diagram: IDiagram) => {
+        return diagram.name !== newName;
+      });
+
+      diagramNumber += 1;
```

Comment:
IMO this is easier to read:

```suggestion
      diagramNumber++;
```